### PR TITLE
Use bitcoincore-rpc client in BitcoinService 

### PR DESCRIPTION
**Pull request by jfldde**
# Description

- [Extract background task out of service creation](https://github.com/chainwayxyz/citrea/commit/ea8da33384845e1f1a78aa600e34a6e746006b0d). Remove side effects from `BitcoinService::new()`
- [Use bitcoin::Network directly](https://github.com/chainwayxyz/citrea/commit/83aba16a008911ffca64ae0968e17a1698f1eba7). Stricter typing
- [Remove Clone bound on DaService](https://github.com/chainwayxyz/citrea/commit/894eeabf54b0c3a1e2883a8cfe544dba13a1d02f). Use of Arc to explicit BitcoinService being shared between thread. 
- [Use bitconcore-rpc client instead of BitcoinNode](https://github.com/chainwayxyz/citrea/commit/715cfdae01978b74590e5ecc64bb0b423249fdb9) Removes the need for custom RPC implementation. Removes `crates/bitcoin-da/src/rpc.rs`
- Future proof BitcoinService RPCs usage. 
- Removes `reqwest` && `base64` dependencies

## Linked Issues
- Pre-requisite for https://github.com/chainwayxyz/citrea/issues/892

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -178,3 +178,4 @@ sha2 = { git = "https://github.com/risc0/RustCrypto-hashes", tag = "sha2-v0.10.8
 # crypto-bigint = { git = "https://github.com/risc0/RustCrypto-crypto-bigint", tag = "v0.5.5-risczero.0" }
 secp256k1_v028 = { package = "secp256k1", version = "0.28", git = "https://github.com/Sovereign-Labs/rust-secp256k1.git", branch = "risc0-compatible-0-28-2" }
 k256 = { package = "k256", version = "0.13.3", git = "https://github.com/risc0/RustCrypto-elliptic-curves", tag = "k256/v0.13.3-risczero.0" }
+bitcoincore-rpc = { version = "0.18.0", git = "https://github.com/jfldde/rust-bitcoincore-rpc.git" }
```
Comment (yaziciahmet) [inline Cargo.toml:181]: my man... don't tell me you are trying to take over the world by installing malware from your local repository.

File: crates/bitcoin-da/src/helpers/parsers.rs [hunk 2]
```diff
@@ -183,7 +183,7 @@ fn parse_relevant_inscriptions(
     })
 }
 
-#[cfg(any(feature = "native", test))]
```
Comment (yaziciahmet) [inline crates/bitcoin-da/src/helpers/parsers.rs:186]: any idea why this was enabled only on native in the first place?

File: crates/bitcoin-da/src/service.rs [hunk 3]
```diff
@@ -69,37 +75,44 @@ const POLLING_INTERVAL: u64 = 10; // seconds
...
             chain_params.reveal_tx_id_prefix,
             tx,
         )
-        .await;
-
-        let serv = this.clone();
+        .await)
+    }
 
+    pub fn spawn_bg_task(
```
Comment (yaziciahmet) [inline crates/bitcoin-da/src/service.rs:106]: Maybe give it a more explicit name? Like `spawn_tx_handler`

File: crates/bitcoin-da/src/service.rs [hunk 4]
```diff
@@ -542,21 +590,49 @@ impl DaService for BitcoinService {
 
     #[instrument(level = "trace", skip(self))]
     async fn get_fee_rate(&self) -> Result<u128, Self::Error> {
-        // This already returns ceil, so the conversion should work
-        let res = self.client.estimate_smart_fee().await? as u128;
+        let smart_fee = self.client.estimate_smart_fee(1, None).await?;
+        let btc_vkb = smart_fee.fee_rate.map_or(0.00001f64, |rate| rate.to_btc());
+
+        let sat_vb_ceil = (btc_vkb * 100_000_000.0 / 1000.0).ceil() as u128;
```
Comment (yaziciahmet) [inline crates/bitcoin-da/src/service.rs:596]: Seeing this for the second time, maybe separate this into it's own method `estimate_smart_fee`?

**Review by yaziciahmet — CHANGES_REQUESTED** [review]

File: crates/bitcoin-da/src/service.rs (hunk 3 above)
Comment (yaziciahmet) [reply crates/bitcoin-da/src/service.rs:106]: For sure

**Review by eyusufatik — APPROVED** [review]

**Review by yaziciahmet — APPROVED** [review]

**Pull request merged**